controllers/limitador_controller.go

```diff
@@ -199,6 +200,16 @@ func (r *LimitadorReconciler) reconcileDeployment(ctx context.Context, limitador
 		return err
 	}
 
+	// TODO: To be deleted when the upgrade path is no longer needed.
```

Comment:
AS the comment says correctly, this is part of the upgrade. So I would add all this code to a function called `UpgradeTov070`. It will be easy to spot in the future and remove it. The implementation of the upgrade is indeed delete the old deployment when the new one is available.

Reply:
Wrapped this into a function in `upgrades` package as suggested.